test/e2e/performanceprofile/functests/utils/tuned/tuned.go

```diff
@@ -84,38 +84,34 @@ func GetPod(ctx context.Context, node *corev1.Node) (*corev1.Pod, error) {
 }
 
 func WaitForStalldTo(run bool, interval, timeout time.Duration, node *corev1.Node) error {
-	return wait.Poll(interval, timeout, func() (bool, error) {
-		cmd := []string{"/bin/bash", "-c", "pidof stalld || true"}
-		stalldPid, err := nodes.ExecCommandOnNode(cmd, node)
-		if err != nil {
-			return false, fmt.Errorf("failed to execute command %q on node: %q; %w", cmd, node.Name, err)
-		}
+	return wait.PollImmediate(interval, timeout, func() (bool, error) {
+		cmd := []string{"/bin/bash", "-c", "pidof stalld"}
+		stalldPid, _ := nodes.ExecCommandOnNode(cmd, node)
 
-		_, err = strconv.Atoi(stalldPid)
+		_, err := strconv.Atoi(stalldPid)
 		if !run { // we don't want stalld to run
 			if err == nil {
-				return false, fmt.Errorf("node=%q stalld_pid=%q stalld is running when it shouldn't", node.Name, stalldPid)
+				return false, nil
 			}
+			return true, nil
```

Comment:
Here it means that `err != nil`. 
It can happens for various reasons if we're not checking that the output received from `ExecCommandOnNode` is OK

Reply:
since we expect `ExecCommandOnNode` command to return error anyway (keep me honest here) then we will need to introduce some error handling here by error type, e.g : if error was for the execution itself or the error came out from not having stalld running

Reply:
> since we expect ExecCommandOnNode command to return error anyway 
No we don't that is my point exactly. if `ExecCommandOnNode` return with an error, you don't know whether it's because stalld is not running (which is sometimes a good thing, meaning the test passes)  or the execution itself failed. 
> need to introduce some error handling here by error type
We don't know what kind of errors might return so we perform such a check efficiently

Reply:
I would argue that that was the exact behavior even before this patch ,  we would still return true at the end of the function and in the last observation we have this was always the case since the syntax with || true was always failing. we can however get the error message from execute command on node and print it out as a warning to provide more debuging info.